### PR TITLE
Fix the usage of a single connection for all database opperations

### PR DESCRIPTION
**Pull request by Kodehawa**
Needs way more testing tho ;-;

### CODE REVIEW

**Comment by Kodehawa** [comment]
The purpose of this is to avoid expensive operations (filtering on an entire table, leaderboards, etc) creating a backlog on the simplest of requests (example a get request) and avoids operations executing in the wrong order by who knows what reason.

This should be fine, but still needs a lot of internal testing to avoid weird stuff happening

File: src/main/java/net/kodehawa/mantarobot/commands/MoneyCmds.java [hunk 1]
```diff
@@ -446,13 +449,18 @@ public void call(GuildMessageReceivedEvent event, String content, String[] args)
...
-                    Cursor<Map> m = r.table("players")
-                            .orderBy()
-                            .optArg("index", r.desc("level"))
-                            .filter(player -> player.g("id").match(pattern))
-                            .map(player -> player.pluck("id", "level"))
-                            .limit(15)
-                            .run(MantaroData.conn(), OptArgs.of("read_mode", "outdated"));
+                    Cursor<Map> m;
+
+                    try(Connection conn = r.connection().hostname(c.dbHost).port(c.dbPort).db(c.dbDb).user(c.dbUser, c.dbPassword).connect()) {
```
Comment (queer) [inline src/main/java/net/kodehawa/mantarobot/commands/MoneyCmds.java]: Make a utility method for getting connections just so you don't keep repeating yourself?

Comment (queer) [reply src/main/java/net/kodehawa/mantarobot/commands/MoneyCmds.java]: Might also be a good idea to pass a callback for a lot of stuff like this just so you don't have to think about remembering the try with resources everywhere

File: src/main/java/net/kodehawa/mantarobot/commands/utils/birthday/BirthdayCacher.java [hunk 2]
```diff
@@ -48,7 +51,11 @@ public BirthdayCacher() {
     public void cache() {
         executorService.submit(() -> {
             try {
-                Cursor<Map> m = r.table("users").run(MantaroData.conn(), OptArgs.of("read_mode", "outdated"));
+                Cursor<Map> m;
+                try(Connection conn = r.connection().hostname(c.dbHost).port(c.dbPort).db(c.dbDb).user(c.dbUser, c.dbPassword).connect()) {
+                    m = r.table("users").run(conn, OptArgs.of("read_mode", "outdated"));
```
Comment (queer) [inline src/main/java/net/kodehawa/mantarobot/commands/utils/birthday/BirthdayCacher.java]: :thinking: :thinking: :thinking: :thinking: :thinking: :thinking: :thinking: :thinking:

Comment (queer) [reply src/main/java/net/kodehawa/mantarobot/commands/utils/birthday/BirthdayCacher.java]: read_mode outdated :thinking:

**Pull request closed without merging**